crates/sui-types/src/traffic_control.rs

```diff
@@ -10,6 +10,22 @@ use std::{fmt::Debug, path::PathBuf};
 
 const TRAFFIC_SINK_TIMEOUT_SEC: u64 = 300;
 
+pub struct Weight(f32);
+
+impl Weight {
+    pub fn new(value: f32) -> Result<Self, &'static str> {
+        if (0.0..=1.0).contains(&value) {
+            Ok(Self(value))
+        } else {
+            Err("Weight must be between 0.0 and 1.0")
```

Comment:
why is this?

Reply:
That's just how the type is constrained. It's just a lever for the service to have generalized control over how errors are weighted in the policy, with the most course control being "include or don't include tallies with this error" as it's being used now, but later on to, for example, say "this error is half as bad as others" with the end result being that its frequency would be weighted by 0.5, requiring then twice as many errors of the same type to lead to being blocklisted.

Not sure if there's an already existing type that could more elegantly represent a float in the range (0..=1) but I could t find any